doc/USER_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+# Installation
+
+To configure a GitHub repository to use TACOS-GHA, you need to configure a
+number of workflows in the `.github/workflows/` directory in the repository.
+Examples of these workflow files can be found [here](workflows).
+
+# Concepts
+
+- slice
+  - A collection of terraform resources sharing a single non-local terraform
+    state backend.
```

Comment:
I think we can either add something here about how slices are built around logical clusters of resources, and/or point to a few examples of slices - like how we've put all the Fastly-specific CDN stuff into the `terraform/internal-sentry/cdn` slice in the ops repo, or how all the Redis-related monitors are kept together in the `datadog/monitors/redis` slice in the datadog-terraform repo.